src/ipc-handlers.ts

```diff
@@ -1081,6 +1070,95 @@ END` );
 	}
 }
 
+export async function openAppAtPath(
+	_event: IpcMainInvokeEvent,
+	editorKey: SupportedEditor,
+	filePath: string
+): Promise< void > {
+	const platform = process.platform;
+	const editor = supportedEditorConfig[ editorKey ];
+
+	if ( ! editor ) {
+		throw new Error( `Editor ${ editorKey } not found or not supported` );
+	}
+
+	if ( platform === 'darwin' ) {
+		return promiseExec( `open -b ${ editor.bundleId } "${ filePath }"` );
+	}
+
+	if ( platform === 'win32' ) {
+		const editorPath = await winFindEditorPath( editorKey );
+		if ( ! editorPath ) {
+			// Fall back to using openURL if no editor path is found
+			return openURL( _event, editor.url( filePath ) );
+		}
+
+		return promiseExec( `"${ editorPath }" "${ filePath }"` );
+	}
+
+	throw new Error( `Platform ${ platform } is not supported` );
+}
+
+/**
+ * Expands Windows environment variables in a path
+ */
+function winExpandPath( path: string ): string {
+	return path.replace( /%([^%]+)%/g, ( _, n ) => process.env[ n ] || '' );
+}
+
+/**
+ * Finds the executable path for a given editor on Windows
+ */
+async function winFindEditorPath( editorKey: SupportedEditor ): Promise< string | null > {
```

Comment:
I would advise moving this function to a module in `src/modules/user-settings/lib`.

Reply:
Good point, thanks. Done in 761352569211bb0534b859ae57bca9e8604ff471.